src/parser/jobs/vpr/modules/RattlingCoil.tsx

```diff
@@ -0,0 +1,110 @@
+import {t} from '@lingui/macro'
+import {Trans, Plural} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {JOBS} from 'data/JOBS'
+import {Event, Events} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+type GaugeModifier = Partial<Record<Event['type'], number>>
+
+const LEFTOVER_COIL_SEVERITY_TIERS = {
+	1: SEVERITY.MINOR,
+	2: SEVERITY.MEDIUM,
+}
+
+const MAX_COILS = 3
+
+export class RattlingCoil extends CoreGauge {
+	static override handle = 'rattlingcoil'
+	static override title = t('vpr.rattlingcoil.title')`Rattling Coil Timeline`
+
+	@dependency private suggestions!: Suggestions
+	@dependency private checklist!: Checklist
+
+	private coilGauge = this.add(new CounterGauge({
+		maximum: MAX_COILS,
+		graph: {
+			handle: 'coils',
+			label: <Trans id="vpr.gauge.resource.coilsLabel">Rattling Coils</Trans>,
+			color: JOBS.VIPER.colour,
+			forceCollapsed: true,
+		},
+	}))
+
+	// Used for Checklist as CounterGauge seems to lack total tracking at this time.
+	private totalGeneratedCoils = 0
```

Comment:
CounterGauge.totalGenerated should give you this value without needing to track it yourself